src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,49 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int LOGIN_MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserDataException("Provided User is null");
+        }
+        validateLogin(user.getLogin());
+        validatePassword(user.getPassword());
+        validateAge(user.getAge());
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException(
+                    "User with such login has been already registered. Login: "
+                    + user.getLogin());
+        }
+        return storageDao.add(user);
```

Comment:
I would first add the user to the database and then return it from it, so you will know for sure that your user was added.

Reply:
When we use add(User user) method from StorageDao class, we have to get added User with id, that User obtained when was added, we have logic with a bug otherwise (this case covered by tests)

Reply:
There will be no error, I just checked that.

Reply:
Look at my logs and what will happen if there is no such user in Storage after the register method execution.

![Screenshot 2023-07-22 203523](https://github.com/mate-academy/jv-registration-validation/assets/113283324/144be8f4-3230-4165-9380-46c0584c5f82)